server/mongo_data_access.go

```diff
@@ -96,19 +154,45 @@ func (dal *mongoDataAccessLayer) Delete(id, resourceType string) error {
 		return convertMongoErr(err)
 	}
 
+	if dal.hasInterceptorsForVerbAndType("DELETE", resourceType) {
+		// Although this is a delete operation we need to get the resource first so we can
+		// run any interceptors on the resource before it's deleted.
+		resource, _ := dal.Get(id, resourceType)
```

Comment:
I agree with that logic - I was thinking about that earlier. Should I make that change?